script/copy_deduplicate

```diff
@@ -358,47 +373,72 @@ def list_live_tables(client, pool, project_id, except_tables, only_tables):
     return [f"{project_id}.{t}" for t in live_tables]
 
 
+class ClientQueue:
+    """Queue for balancing jobs across billing projects."""
+
+    def __init__(self, billing_projects, parallelism):
+        clients = [bigquery.Client(project) for project in billing_projects]
+        self._q = Queue(parallelism)
+        for i in range(parallelism):
+            self._q.put(clients[i % len(clients)])
+
+    @contextmanager
+    def client(self):
+        client = self._q.get_nowait()
+        try:
+            yield client
+        finally:
+            self._q.put_nowait(client)
+
+    def with_client(self, func, *args):
+        with self.client() as client:
+            func(client, *args)
+
+
 def main():
     args = parser.parse_args()
 
-    client = bigquery.Client()
+    # create a queue for balancing load across projects
+    client_q = ClientQueue(args.billing_projects, args.parallelism)
 
     with ThreadPool(args.parallelism) as pool:
-        live_tables = list_live_tables(
-            client=client,
-            pool=pool,
-            project_id=args.project_id,
-            except_tables=args.except_tables,
-            only_tables=args.only_tables,
-        )
-
-        job_args = [
-            args
-            for jobs in pool.starmap(
-                get_query_job_configs,
-                [
-                    (
-                        client,
-                        live_table,
-                        args.date,
-                        args.dry_run,
-                        args.slices,
-                        args.priority,
-                        args.preceding_days,
-                    )
-                    for live_table in live_tables
-                ],
+        with client_q.client() as client:
```

Comment:
These two context managers can be combined in one statement to reduce nesting. I think the syntax is:

    with ThreadPool(args.parallelism) as pool, client_q.client() as client:

Reply:
Or maybe they have different lifetimes here.

Reply:
they have different lifetimes